### PR TITLE
Make hello component a factory

### PR DESCRIPTION
**Pull request by troutowicz**
Using component factories prevents problematic multiple React instance scenarios.

### CODE REVIEW

File: source/hello.js [hunk 1]
```diff
@@ -1,14 +1,14 @@
-import React from 'react';
+import PropTypes from 'prop-types';
```
Comment (koresar) [inline source/hello.js:1]: This file needs `/* global React */` too. Right?

Comment (troutowicz) [inline source/hello.js:1]: No. ;) React is passed in.

Comment (koresar) [inline source/hello.js:1]: Right. :)

Comment (ericelliott) [inline source/hello.js:1]: Gotta love that new ES6 syntax! =)

**Pull request merged**